cpp/src/parquet/stream_writer.h

```diff
@@ -0,0 +1,135 @@
+#ifndef PARQUET_STREAM_WRITER_H
+#define PARQUET_STREAM_WRITER_H
+
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "parquet/api/writer.h"
+
+namespace parquet {
+
+// A class for writing Parquet files using an output stream type API.
+//
+// The values given must be of the correct type i.e. the type must
+// match the file schema exactly otherwise a ParquetException will be
+// thrown.
+//
+// The user must explicitly advance to the next row using the
+// nextRow() function.
+//
+// A maximum row group size can be configured, the default size is 512
+// Mio.  Alternatively the row group size can be set to zero and the
```

Comment:
Mio? Is the requirement either or?  is 512 standard in the code base, that seems on the small side.

Reply:
https://parquet.apache.org/documentation/latest/ recommends the row group size be between 512 MB and 1GB
As for "Mio" is a "mebioctet" which is commonly used in Europe.  Can change to the old "MB" units if that's preferred.